src/reference/03-Developers-Guide/02-Coding-Guideline/00.md

```diff
@@ -0,0 +1,204 @@
+---
+out: Coding-Guideline.html
+---
+
+  [jsuereth2012]: http://jsuereth.com/scala/2012/04/22/scaladays.html
+  [pins_trait127]: http://www.artima.com/pins1ed/traits.html#12.7
+  [mima]: https://github.com/typesafehub/migration-manager
+
+sbt Coding Guideline
+--------------------
+
+This page discusses the coding style and other guidelines for sbt 1.0.
+
+### General goal
+
+sbt 1.0 will primarily target Scala 2.11.
+We will cross build against Scala 2.10.
+
+#### Clean up old deprecation
+
+Before 1.0 is release, we should clean up deprecations.
+
+#### Aim for zero warnings (except deprecation)
+
+On Scala 2.11 we should aim for zero warnings.
+One exception may be deprecation if it's required for 2.10 cross building.
+
+### Modular design
+
+#### Aim small
+
+The fewer methods we can expose to the build user, the easier it becomes to maintain it.
+
+#### Public APIs should be coded against "interfaces"
+
+Code against interface.
+
+#### Hide implementation details
+
+The implementation details should be hidden behind `sbt.internal.x` package,
+where `x` could be the name of the main package (like `io`).
+
+#### Depend less
+
+Making independent modules with fewer dependent libraries make it easier to reuse them.
+
+#### Hide external classes
+
+Avoid exposing external classes out to API, except for standard Scala and Java classes.
+
+#### Hide internal modules
+
+A module may be declared internal if there's no use for public.
```

Comment:
We should note: that you should not depend on another module's internals (unless you must).